### PR TITLE
fix: provider in external-dns

### PR DESCRIPTION
**Pull request by j-zimnowoda**

### CODE REVIEW

File: values/external-dns/external-dns.gotmpl [hunk 1]
```diff
@@ -1,20 +1,22 @@
...
 ## - aws, azure, cloudflare, coredns, designate, digitalocoean, google, infoblox, rfc2136
-provider: {{ $dnsProvider }}
-{{- if eq $dnsProvider "google" }}
+
+{{- if hasKey $dns "google" }}
+provider: "google"
 google:
   project: {{ $dns.google.projectId }}
   serviceAccountKey: |
     {{- $dns.google.cloudDnsKey | nindent 4 }}
```
Comment (Morriz) [inline values/external-dns/external-dns.gotmpl:14]: can we maybe normalize this and use the property names that are asked by external-dns? Just like for aws and azure? Then we can just do
```yaml
google:  {{- $dns.google | toYaml | nindent 2 }}
```

File: values/external-dns/external-dns.gotmpl [hunk 2]
```diff
@@ -1,20 +1,22 @@
...
-{{- else if eq $dnsProvider "aws" }}
+{{- else if hasKey $dns "aws" }}
+provider: "aws"
 aws: {{- $dns.aws | toYaml | nindent 2 }}
-{{- else if eq $dnsProvider "azure" }}
+{{- else if hasKey $dns "azure" }}
+provider: "azure"
 azure:
   {{- $dns.azure | toYaml | nindent 2 }}
```
Comment (Morriz) [inline values/external-dns/external-dns.gotmpl:21]: can be on one line

**Review by Morriz — CHANGES_REQUESTED** [review]

File: values/external-dns/external-dns.gotmpl (hunk 1 above)
Comment (j-zimnowoda) [reply values/external-dns/external-dns.gotmpl:14]: yes, that's good idea

Comment (j-zimnowoda) [reply values/external-dns/external-dns.gotmpl]: done

**Review by Morriz — APPROVED** [review]

**Pull request merged**